backend/chat/active-chatters.js

```diff
@@ -29,6 +30,15 @@ function addOrUpdateActiveChatter(user) {
         return;
     }
 
+    // Stop early if user shouldn't be in active chatter list.
+    userDatabase.getUserByUsername(user.user_name).then(userDB => {
+        if (userDB.disableActiveUserList) {
+            logger.debug(userDB.username + " is set to not join the active viewer list.");
+            return;
```

Comment:
I think there is a bug here. The user will still get added to the active list because when you `return` here, you are only returning in the `then()` function, not the outer `addOrUpdateActiveChatter` function. So execution of `addOrUpdateActiveChatter` isn't stopped. 

I would suggest making `addOrUpdateActiveChatter` async:
```js
asyn function addOrUpdateActiveChatter(user) {
```

and then await the `getUser` call:
```js
let firebotUser = await userDatabase.getUserByUsername(user.user_name);
if (firebotUser != null && firebotUser.disableActiveUserList) {
            logger.debug(firebotUser.username + " is set to not join the active viewer list.");
            return;
}
```

Reply:
Going forward I think we should default to using `async/await` on promises vs doing `.then()` chaining. I think async/await is easier to read (especially compared to being chained like three .then()s deep) and it inherently helps avoid asynchronous bugs like this. 

I know `async/await` stuff is newer and a little weird at first but perhaps after a few repetitions, it'll come more naturally.